actors/builtin/paych/paych_test.go

```diff
@@ -0,0 +1,200 @@
+			WithActorType(newPaychAddr, builtin.AccountActorCodeID).
+			WithActorType(payerAddr, builtin.AccountActorCodeID)
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt, payerAddr, newPaychAddr)
+		var st State
+		rt.GetState(&st)
+		assert.Equal(t, newPaychAddr, st.To)
+		assert.Equal(t, payerAddr, st.From)
+		assert.Empty(t, st.LaneStates)
+	})
+
+	testCases := []struct {
+		desc string
+		newActorAddr addr.Address
+		callerCode cid.Cid
+		newActorCode cid.Cid
+		payerCode cid.Cid
+		expExitCode exitcode.ExitCode
+
+	} {
+		{ "fails if target (to) is not account actor",
+			newPaychAddr,
+			builtin.InitActorCodeID,
+			builtin.CronActorCodeID,
+			builtin.AccountActorCodeID,
+			exitcode.ErrIllegalArgument,
+		},{ "fails if sender (from) is not account actor",
+			newPaychAddr,
+			builtin.InitActorCodeID,
+			builtin.CronActorCodeID,
+			builtin.AccountActorCodeID,
+			exitcode.ErrIllegalArgument,
+		},{ "fails if addr is not ID type",
+			tutil.NewActorAddr(t, "beach blanket babylon"),
+			builtin.InitActorCodeID,
+			builtin.CronActorCodeID,
+			builtin.AccountActorCodeID,
+			exitcode.ErrIllegalArgument,
+		},
+	}
+	for _,tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			builder := mock.NewBuilder(ctx, newPaychAddr).
+				WithCaller(callerAddr, tc.callerCode).
+				WithActorType(newPaychAddr, tc.newActorCode).
+				WithActorType(payerAddr, tc.payerCode)
+			rt := builder.Build(t)
+			rt.ExpectValidateCallerType(builtin.InitActorCodeID)
+			rt.ExpectAbort(tc.expExitCode, func() {
+				rt.Call(actor.Constructor, &ConstructorParams{To: newPaychAddr})
+			})
+		})
+	}
+}
+
+func TestPaymentChannelActor_UpdateChannelState(t *testing.T) {
+	ctx := context.Background()
+	actor := pcActorHarness{Actor{}, t}
+
+	newPaychAddr := tutil.NewIDAddr(t, 100)
+	callerAddr := tutil.NewIDAddr(t, 101)
+	payerAddr := tutil.NewIDAddr(t, 102)
+
+	// set up trivial signature verifier and hasher
+	versig := func(sig crypto.Signature, signer addr.Address, plaintext []byte) bool { return true }
+	hasher := func(data []byte) []byte { return data }
+
+	balance:= abi.NewTokenAmount(100)
+	received := abi.NewTokenAmount(0)
+	builder := mock.NewBuilder(ctx, newPaychAddr).
+		WithBalance(balance, received).
+		WithEpoch(abi.ChainEpoch(2)).
+		WithCaller(callerAddr, builtin.InitActorCodeID).
+		WithActorType(newPaychAddr, builtin.AccountActorCodeID).
+		WithActorType(payerAddr, builtin.AccountActorCodeID).
+		WithVerifiesSig(versig).
+		WithHasher(hasher)
+
+
+	t.Run("Can add a lane/update actor state", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt, payerAddr, newPaychAddr)
+		amt := big.NewInt(10)
+		lane := int64(999)
+		nonce := int64(1)
+		sig := &crypto.Signature{
+			Type: crypto.SigTypeBLS,
+			Data: []byte("doesn't matter"),
+		}
+		tl := abi.ChainEpoch(1)
+		sv := SignedVoucher{
+			TimeLock:  tl,
+			Lane:      lane,
+			Nonce:     nonce,
+			Amount:    amt,
+			Signature: sig,
```

Comment:
Nit: I suggest inlining some of these vars/fields for brevity since they can still be referenced by good names like `sv.Amount` further down.